src/components/NavBar/ThemeSwitch.jsx

```diff
@@ -0,0 +1,45 @@
+import DarkModeIcon from '@mui/icons-material/DarkMode';
+import LightModeIcon from '@mui/icons-material/LightMode';
+import { IconButton } from '@mui/material';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+    getTheme,
+    toggleDark,
+    toggleLight,
+} from '../../redux/slices/themeSlice.js';
+
+const ThemeSwitch = () => {
+    const dispatch = useDispatch();
+    const theme = useSelector((state) => getTheme(state));
+
+    const handleThemeToggle = () => {
+        if (theme.palette.mode === 'light') {
```

Comment:
You can use shortened syntax: `dispatch(theme.palette.mode === 'light' ? toggleDark() : toggleLight())`